File: java/com/countrygamer/angel/common/entity/WAEntity.java
package com.countrygamer.angel.common.entity;

import net.minecraft.entity.EntityList;
import net.minecraft.entity.EnumCreatureType;
import net.minecraft.world.biome.BiomeGenBase;

import com.countrygamer.angel.common.WAOptions;
import com.countrygamer.angel.common.WeepingAngels;
import com.countrygamer.core.Base.Plugin.PluginEntityRegistry;

import cpw.mods.fml.common.registry.EntityRegistry;

public class WAEntity implements PluginEntityRegistry {
	
	@Override
	public void registerEntities() {
		EntityRegistry.registerModEntity(EntityWeepingAngel.class, "Weeping Angel", 1,
				WeepingAngels.instance, 80, 3, false);
		
	}
	
	@Override
	public void addEntityMappings() {
		EntityList.addMapping(EntityWeepingAngel.class, "WeepingAngel", this.getNewEntityID(),
				0x808080, 0xD1D1D1);
		
	}
	
	@Override
	public void registerEntitySpawns() {
		if (WAOptions.spawnRate_Angel > 0) {
			for (int i = 0; i < BiomeGenBase.getBiomeGenArray().length; i++) {
				if (BiomeGenBase.getBiomeGenArray()[i] != null) {
					// ~ mob file, spawn frequency, group size,
					// ~ creature type, biome
					cpw.mods.fml.common.registry.EntityRegistry.addSpawn(EntityWeepingAngel.class,
							WAOptions.spawnRate_Angel, 1, WAOptions.maxSpawnPerInstance_Angel,
							EnumCreatureType.monster, BiomeGenBase.getBiomeGenArray()[i]);
				}
			}
		}
		
	}
	
	public int getNewEntityID() {
		int id = 0;
		while (EntityList.getStringFromID(id) != null)
			id++;
		return id;
	}
	
}

File: src/main/scala/com/countrygamer/weepingangels/common/init/WAEntity.scala
package com.countrygamer.weepingangels.common.init

import com.countrygamer.cgo.wrapper.common.registries.EntityRegister
import com.countrygamer.weepingangels.common.entity.EntityWeepingAngel
import com.countrygamer.weepingangels.common.{WAOptions, WeepingAngels}
import cpw.mods.fml.common.registry.EntityRegistry
import net.minecraft.entity.EnumCreatureType
import net.minecraft.world.biome.BiomeGenBase

/**
 *
 *
 * @author CountryGamer
 */
object WAEntity extends EntityRegister {

	override def register(): Unit = {
		EntityRegistry
				.registerModEntity(classOf[EntityWeepingAngel], "Weeping Angel", 1, WeepingAngels,
		            80, 3, false)

	}

	override def addEntityMappings: Unit = {
		EntityRegistry.registerGlobalEntityID(classOf[EntityWeepingAngel], "Weeping Angel",
			this.getNewEntityID(), 0x808080, 0xD1D1D1)

	}

	override def addEntitySpawns: Unit = {
		var index: Int = 0
		for (index <- 0 until BiomeGenBase.getBiomeGenArray.length) {
			val biome: BiomeGenBase = BiomeGenBase.getBiomeGenArray()(index)

			if (biome != null && !(biome == BiomeGenBase.mushroomIsland ||
					biome == BiomeGenBase.mushroomIslandShore)) {
				EntityRegistry
						.addSpawn(classOf[EntityWeepingAngel], WAOptions.spawnProbability, 1, 1,
				            EnumCreatureType.monster, biome)

			}

		}

	}

}
